lib/lotus/application_name.rb

```diff
@@ -0,0 +1,95 @@
+module Lotus
+  # An application name.
+  #
+  # @since x.x.x
+  class ApplicationName
+
+    # A list of words that are prohibited from forming the application name
+    #
+    # @since x.x.x
+    INVALID_WORDS = %w(lotus).freeze
+
+
+    # Initialize and check against reserved words
+    #
+    # An application name needs to be translated in quite a few ways:
+    # First, it must be checked against a list of reserved words and rejected
+    # if it is invalid. Secondly, assuming it is not invalid, it must be able
+    # to be output roughly as given, but with the following changes:
+    #
+    # 1. downcased,
+    # 2. with surrounding spaces removed,
+    # 3. with internal whitespace rendered as underscores
+    # 4. with underscores de-duplicated
+    #
+    # which is the default output:
+    #
+    #     ApplicationName.new("my-App ").to_s # => "my-app"
+    #
+    # It must also be transformable into an environment variable:
+    #
+    #     ApplicationName.new("my-app").to_env_s => "MY_APP"
+    #
+    # @return [Lotus::ApplicationName] a new instance of the application name
+    #
+    # @since x.x.x
+    def initialize(name)
+      @name = sanitize(name)
+      ensure_validity!
+    end
+
+
+    # Returns the cleaned application name.
+    #
+    # @since x.x.x
+    def to_s
+      @name
+    end
+
+
+    # Returns the application name uppercased with non-alphanumeric characters
+    # as underscores.
+    #
+    # @since x.x.x
+    def to_env_s
+      @name.upcase.gsub(/\W/, '_')
+    end
+
+
+    # Returns true if a potential application name matches one of the reserved
+    # words.
+    #
+    # @since x.x.x
+    def self.invalid?(name)
+      INVALID_WORDS.include?(name)
+    end
+
+
+    private
+
+    # Raises RuntimeError with explanation if the provided name is invalid.
+    #
+    # @api private
+    # @since x.x.x
+    def ensure_validity!
+      if self.class.invalid?(@name)
+        raise RuntimeError,
```

Comment:
I don't think this is a problem.
This exception isn't meant to be handled by developers, it's only a way to quit the CLI application with a non-successful state.

Indeed it could be optionally expressed as:

```ruby
fail "application name.."
```